src/QsCompiler/QirGeneration/DebugInfoManager.cs

```diff
@@ -318,13 +317,12 @@ internal IrFunction CreateGlobalFunction(QsSpecialization spec, string mangledNa
                         linePosition: ConvertToDebugPosition(absolutePosition),
                         signature: debugSignature,
                         isLocalToUnit: true, // We're using the compile unit from the source file this was declared in
-                        isDefinition: isDefinition,
+                        isDefinition: true, // if this isn't set to true, it results in temporary metadata nodes that don't get resolved.
                         scopeLinePosition: ConvertToDebugPosition(absolutePosition), // TODO: Could make more exact bc of formatting and white space (see lastParamLocation in Kaleidescope tutorial)
                         debugFlags: DebugInfoFlags.None,
                         isOptimized: false,
                         curDIBuilder);
 
-                    this.EmitLocation(absolutePosition, func.DISubProgram);
```

Comment:
Why invoking `EmitLocation` is no longer needed?

Reply:
It was redundant with other debug locations, and the scoping was slightly off. This emitted a debug location wherever the function is declared, yet set the scope to the function itself. The debug locations should be set within the scope of the function only when steps are executed within the function. And then they should be set within the scope of the calling function when this function is called. Those situations are taken care of elswhere.